GitUI/Editor/Diff/DiffLineNumAnalyzer.cs

```diff
@@ -13,7 +13,7 @@ public class DiffLineNumAnalyzer
 
         public DiffLinesInfo Analyze(string diffContent)
         {
-            var ret = new DiffLinesInfo();
+            DiffLinesInfo ret = new();
             var isCombinedDiff = PatchProcessor.IsCombinedDiff(diffContent);
             var lineNumInDiff = 0;
             var leftLineNum = DiffLineInfo.NotApplicableLineNum;
```

Comment:
Discussion point: For projects that prefer var, I think a change like this makes the code look a bit inconsistent, as the variables no longer line up. It's a minor point, and of course any code style can be learned, but it does seem a bit harder to read with this change.

I feel much more strongly about the case where the type name is completely removed (e.g. `foo.Bar(new())`), but that does not appear to be present in this PR.

Reply:
I like the changes in e1168fb4885c7c1d0b649c972551a71c7aa2593a.

Reply:
I don't like `var` because more often than not it makes the code opaque (e.g. `var result = GetSomething();`), and prefer to avoid it.
The new `new` syntax makes it better in my opinion.

Also I think I rather like `foo.Bar(new())` instead of `foo.Bar(new Some.CrazyLongTypeName.ThatIDontCareAbout())` because it provides enough information for me. Obviously, like with anything else, a common sense must be applied as this may become unreadable `foo.Bar(new(), new(new, new()), new())`

Reply:
It seems contradictory to dislike var due to invisible types, while also preferring new Foo(new())). Am I missing something?

One issue with new Foo(new())) is that if you add an overload of Foo, your code no longer compiles.

Reply:
> It seems contradictory to dislike var due to invisible types, while also preferring new Foo(new())). Am I missing something?

I find something like the following fine, then the type is "close".
```c#
List<DateTime> l = new() { new() };
```

> One issue with new Foo(new())) is that if you add an overload of Foo, your code no longer compiles.

Need to be handled by the designer adding the overload or updating a library.

But I do not want to use `new()` much more than in this PR or really remove `var` generally.

Reply:
> It seems contradictory to dislike var due to invisible types, while also preferring new Foo(new())). Am I missing something?

I see no contradictions here.
I want to see what type is being returned from a call so I can reason about how it is used. But when I'm see a type constructed I don't _really_ care what goes in (the compiler will check).